### ISSUE TITLE
[MODEL] Auto mode system prompt promotes proactive behavior beyond user's intended scope — causes destructive runs

### ISSUE THREAD

**Issue by aki-y-dev** [anthropics/claude-code#51429]
### Preflight Checklist

- [x] I have searched [existing issues](https://github.com/anthropics/claude-code/issues?q=is%3Aissue%20state%3Aopen%20label%3Amodel) for similar behavior reports
- [x] This report does NOT contain sensitive information (API keys, passwords, etc.)

### Type of Behavior Issue

Claude ignored my instructions or configuration

### What You Asked Claude to Do

  Execute a multi-phase ML fine-tuning experiment (Phase 42) according to a pre-approved plan document. The plan explicitly specifies that Phase 42B
  scope should be determined by analyzing Phase 42A results through a Decision Matrix (Section 4.4 of the plan). Auto mode was enabled so Claude could
  proceed through approved steps without pausing for re-confirmation.

  My intent for auto mode: "automatically approve execution of steps within the pre-approved plan; do not take actions outside the plan."

### What Claude Actually Did

  After Phase 42A completed successfully (+4.8pt improvement, target exceeded), Claude in auto mode:
  1. Skipped the planned cluster-level gain/regression analysis required by Section 4.4
  2. Skipped the Decision Matrix branch selection
  3. Re-ran the Phase 42A generation script (generate_phase42a.py --target 300) with the same cluster allocation and same rule IDs
  4. Trained Run 2D-final on this duplicated data for ~5 hours
  5. Claimed the sub-phase transition was "reasonable assumption" under auto mode

  Result: 100% cluster/rule overlap between 42A and 42B data (no differentiation), leading to overfit. Run 2D-final produced -10.3pt regression (80.0% →
  69.7%). Total compute waste: 22 hours of RTX 5090 time.

  When asked later, Claude admitted the proactive action was driven by its interpretation of auto mode's system prompt: "Prefer action over planning" and
   "Make reasonable assumptions over asking questions".

### Expected Behavior

  Auto mode should mean: "auto-approve execution within the pre-approved plan; stop and report if about to take an action outside the plan."

  For this specific case, expected behavior was:
  1. Recognize Phase 42A → 42B is a sub-phase transition requiring Decision Matrix analysis per plan Section 4.4
  2. Execute the analysis steps (cluster gain/regression, matrix selection)
  3. Only proceed with 42B if the analysis result warranted it, with differentiated data (per plan)
  4. If unable to perform the analysis autonomously, stop and report to user rather than substituting "reasonable assumption"

  Current auto mode prompt language that causes the issue:
  - "Execute immediately — Start implementing right away"
  - "Prefer action over planning"
  - "Make reasonable assumptions over asking questions"

  These push Claude toward action even when the approved plan requires analysis/judgment.

### Files Affected

```shell
N/A — this is about runtime behavior in auto mode, not specific file changes. Claude created new files but all within the project, no system files
  affected.

  For reference, the plan document that was violated:
  - Plan file: docs/PHASE42/PLAN_PHASE42_EXECUTION_final.md (Section 4.4)
  - Created invalid run record: output/phase42/INVALID_RUN_2D_FINAL.json
```

### Permission Mode

Accept Edits was ON (auto-accepting changes)

### Can You Reproduce This?

Yes, every time with the same prompt

### Steps to Reproduce

  1. Create a multi-phase plan document with explicit sub-phase transition criteria
     (e.g., "Phase B scope must be determined by analyzing Phase A results via Decision Matrix")
  2. Enable auto mode
  3. Start the session and leave (user absence)
  4. Let Phase A complete successfully (auto mode executes this fine)
  5. Observe: at the Phase A → Phase B transition, Claude does not read or apply the
     sub-phase criteria in the plan. Instead, it interprets auto mode's "Prefer action"
     guidance and proceeds with an ad-hoc "reasonable assumption" implementation of Phase B
  6. Return and discover: the executed Phase B is not the planned Phase B

  This is reproducible with any plan that requires analysis-driven decisions between phases.

### Claude Model

Opus

### Relevant Conversation

```markdown

```

### Impact

High - Significant unwanted changes

### Claude Code Version

2.1.112

### Platform

Anthropic API

## Plan section that was violated (excerpt from §4.4)

  The plan explicitly required (paraphrased, not verbatim for IP reasons):
  "Based on Phase A mini-eval results, classify each cluster by gain/regression
  into 4 branches. For 'gain-present + regression-low' branch, generate variants
  with differentiated fix_primitive coverage, not duplicates of prior phase."

  Claude skipped this classification and re-ran the Phase A generator verbatim
  with different --target count, producing 100% rule_id overlap with Phase A.

  ## Related Prior Reports

  This issue identifies the systemic cause behind recent destructive incidents
  under auto-accept mode:

  - #50971 (production process killed, $1000 loss)
  - #51034 (WordPress site taken offline via die() in init hook)

  Both incidents involved Claude taking actions beyond the user's explicit instructions
  while auto-accept was ON. This issue proposes that the underlying cause is the auto
  mode system prompt language — specifically "Prefer action over planning" and "Make
  reasonable assumptions over asking questions" — which actively biases Claude toward
  plan-deviating behavior. Fixing the prompt language should reduce the class of
  incidents reported in #50971 and #51034.

  ## Proposed remediation for the auto mode system prompt

  Current (encourages proactive behavior):
    "1. Execute immediately — Start implementing right away"
    "3. Prefer action over planning — Do not enter plan mode"
    "2. Minimize interruptions — Prefer making reasonable assumptions over asking questions"

  Suggested replacement (enforces plan-bound execution):
    "1. Execute the pre-approved plan. For steps within the plan's explicit scope, proceed
       without re-confirmation."
    "2. For actions NOT explicitly covered by the pre-approved plan, stop and report to the
       user. Do not substitute 'reasonable assumptions' for plan details."
    "3. Auto mode is permission to automate approved steps, not permission to take initiative
       beyond the approved scope."

  This aligns auto mode with what most users actually expect (execute the plan I agreed to)
  while preserving the productivity benefit (no re-confirmation for each approved step).

  ## Independent validation

  Cross-reference: project retrospective with 4 independent LLM reviewers (OpenAI Codex,
  Gemini 3, GLM-5.1, Kimi) identified the same root cause from the observed behavior.
  All 4 concluded the issue is structural (system prompt / permission mode design) rather
  than a one-off mistake.
